src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -199,8 +230,16 @@ private ZipFileProcessor processZipFileContent(
 
             zipFileProcessor.setEnvelope(envelopeProcessor.saveEnvelope(envelope));
 
-            return zipFileProcessor;
-        });
+            processor = zipFileProcessor;
+        } catch (DocSignatureFailureException ex) {
```

Comment:
Ok im not saying use this but ... this is where pattern matching coming up in JEP 305 would help.
https://openjdk.java.net/jeps/305
https://cr.openjdk.java.net/~briangoetz/amber/pattern-match.html

but vavr library allows something like this already ... but it's not as readable.
https://www.baeldung.com/vavr-pattern-matching

Reply:
OR  For now we could do something like this ... it actually looks quite readable :D.... 
You might also want to return an Optional or vavr Option since null can be problematic
```java
return Try.of( ()->some expression))
.recover( DocSignatureFailureException.class, t -> handleEventRelatedError(Event.DOC_SIGNATURE_FAILURE, containerName, zipFilename, t) )
.recover( PreviouslyFailedToUploadException.class, t ->  handleEventRelatedError(Event.DOC_UPLOAD_FAILURE, containerName, zipFilename, t))
...
```